functions/src/modules/Users.functions.ts

```diff
@@ -25,27 +24,40 @@ export const processRegistration = functions.database
     return false;
   }
 
+  const _services = {
```

Comment:
Using an underscore is not a standard (AFAIK). Please rename to `serviceRoles`.
And why not to inverse the dictionary? It would simplify logic below. It would become just `result[serviceRoles[service]] = true;`. Isn't it? The only thing is to check for existence in a similar way as you do.